assets/app/scripts/services/data.js

```diff
@@ -199,15 +125,269 @@ angular.module('openshiftConsole')
   };
 
 
-  DataService.prototype._stopListeningForUpdates = function(type) {
-    if (this._openConnections[type]) {
-     this._openConnections[type].close();
-     // TODO can we use delete here instead, or will that screw up the onclose event
-     this._openConnections[type] = null;
+  function DataService() {
+    this._listCallbacksMap = {};
+    this._watchCallbacksMap = {};
+    this._watchOperationMap = {};
+    this._listOperationMap = {};
+    this._resourceVersionMap = {};
+    this._dataMap = {};
+    this._watchWebsocketsMap = {};
+    this._watchPollTimeoutsMap = {};
+  }
+
+// type:      API type (e.g. "pods")
+// context:   API context (e.g. {project: "..."})
+// callback:  function to be called with the initial list of the requested type,
+//            and when updates are received, parameters passed to the callback:
+//            list:   current list of objects (context-qualified)
+//                    represented as a list (to preserve server-order)
```

Comment:
update comment, no longer a list or directly usable (all access goes through the `by` helper)